src/control/cad.js

```diff
@@ -253,103 +261,265 @@ class CadControl extends Control {
+
+    // Use intersection points of the four lines to get the extent
+    const intersectTopLeft = OverlayOp.intersection(
+      parser.read(minVertLine),
+      parser.read(minHoriLine),
+    );
+    const intersectBottomLeft = OverlayOp.intersection(
+      parser.read(minVertLine),
+      parser.read(maxHoriLine),
+    );
+    const intersectTopRight = OverlayOp.intersection(
+      parser.read(maxVertLine),
+      parser.read(minHoriLine),
+    );
+    const intersectBottomRight = OverlayOp.intersection(
+      parser.read(maxVertLine),
+      parser.read(maxHoriLine),
+    );
+
+    return [
+      [intersectTopLeft.getCoordinate().x, intersectTopLeft.getCoordinate().y],
+      [
+        intersectBottomLeft.getCoordinate().x,
+        intersectBottomLeft.getCoordinate().y,
+      ],
+      [
+        intersectTopRight.getCoordinate().x,
+        intersectTopRight.getCoordinate().y,
+      ],
+      [
+        intersectBottomRight.getCoordinate().x,
+        intersectBottomRight.getCoordinate().y,
+      ],
+    ];
+  }
+
   /**
    * Draws snap lines by building the extent for
    * a pair of features.
    * @private
    * @param {Array.<ol.Feature>} features List of features.
    * @param {ol.Coordinate} coordinate Mouse pointer coordinate.
+   * @param {Array.<number>} pixel Mouse pointer pixel array.
    */
-  drawSnapLines(features, coordinate) {
+  drawSnapLines(features, coordinate, pixel) {
     let auxCoords = [];
 
+    // First get all snap points: neighbouring feature vertices and extent corners
     for (let i = 0; i < features.length; i += 1) {
       const geom = features[i].getGeometry();
       const featureCoord = geom.getCoordinates();
-
-      if (geom instanceof Point) {
-        auxCoords.push(featureCoord);
-      } else {
-        // filling snapLayer with features vertex
-        if (geom instanceof LineString) {
-          for (let j = 0; j < featureCoord.length; j += 1) {
-            auxCoords.push(featureCoord[j]);
-          }
-        } else if (geom instanceof Polygon) {
-          for (let j = 0; j < featureCoord[0].length; j += 1) {
-            auxCoords.push(featureCoord[0][j]);
+      if (featureCoord.length) {
+        if (geom instanceof Point) {
+          auxCoords.push(featureCoord);
+        } else {
+          // Add feature vertices
+          if (geom instanceof LineString) {
+            for (let j = 0; j < featureCoord.length; j += 1) {
+              auxCoords.push(featureCoord[j]);
+            }
+          } else if (geom instanceof Polygon) {
+            for (let j = 0; j < featureCoord[0].length; j += 1) {
+              auxCoords.push(featureCoord[0][j]);
+            }
           }
-        }
 
-        // filling auxCoords
-        const coords = fromExtent(geom.getExtent()).getCoordinates()[0];
-        auxCoords = auxCoords.concat(coords);
+          // Add extent corner coordinates
+          const coords = this.getRotatedExtent(geom);
```

Comment:
Why do we need a rotated extent ?

Reply:
Yes so the bbox extent around the feature orients itself to the current view, not geographical north and south